enterprise/backend/test/metabase_enterprise/data_editing/action_v2_api_test.clj

```diff
@@ -29,99 +29,98 @@
+                action-kind->expected-params {"row/create" expected-row-params
+                                              "row/update" (concat expected-id-params expected-row-params)
+                                              "row/delete" expected-id-params}]
 
-          (doseq [[action-kind expected-params] action-kind->expected-params]
-            (mt/with-actions [model {:type          :model
-                                     :dataset_query (mt/mbql-query orders)}
-                              {action-id :action-id} {:type :implicit
-                                                      :kind action-kind
-                                                      :name "Do cool thing"}]
-              (is (= {:title      "Do cool thing"
-                      :parameters expected-params}
-                     (mt/user-http-request :crowberto :post 200 "action/v2/configure"
-                                           {:scope     {:model-id (:id model)
-                                                        :table-id (mt/id :orders)}
-                                            :action_id {:action-id action-id}}))))))))))
+            (doseq [[action-kind expected-params] action-kind->expected-params]
+              (mt/with-actions [model {:type          :model
+                                       :dataset_query (mt/mbql-query orders)}
+                                {action-id :action-id} {:type :implicit
+                                                        :kind action-kind
+                                                        :name "Do cool thing"}]
+                (is (= {:title      "Do cool thing"
+                        :parameters expected-params}
+                       (mt/user-http-request :crowberto :post 200 "action/v2/configure"
+                                             {:scope     {:model-id (:id model)
+                                                          :table-id (mt/id :orders)}
+                                              :action_id {:action-id action-id}})))))))))))
 
 ;; For example, when picking a table action to add as a row action, we call this for the initial config form.
 (deftest configure-table-action-test
   (mt/with-premium-features #{:table-data-editing}
     (mt/test-drivers #{:h2 :postgres}
-      (data-editing.tu/toggle-data-editing-enabled! true)
-      (with-open [test-table (data-editing.tu/open-test-table! {:id        'auto-inc-type
-                                                                :text      [:text]
-                                                                :int       [:int]
-                                                                :timestamp [:timestamp]
-                                                                :date      [:date]}
-                                                               {:primary-key [:id]})]
+      (data-editing.tu/with-test-tables! [test-table [{:id        'auto-inc-type
```

Comment:
FYI the reason we had the ref before is because @wotbrew is allergic to the macro creep.

What's really nice with returning a bunch of closeable things from functions is that we can mix a whole bunch of different types of setup at the same level of indentation, avoiding the indentation creep our test typically suffer from...

That said, I hated having to deref that table id, and I'm ok with one more macro myself... one step closer to the `with-with-macros` macro 😬